### ISSUE TITLE
[Test Coverage] Implement Tests for Classification Loss Functions

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#346]
## Problem
Classification fitness calculators in `src/FitnessCalculators/` have 0% test coverage.

## Files Requiring Tests (5 files)
- BinaryCrossEntropyLossFitnessCalculator.cs
- CategoricalCrossEntropyLossFitnessCalculator.cs
- CrossEntropyLossFitnessCalculator.cs
- HingeLossFitnessCalculator.cs
- FocalLossFitnessCalculator.cs

## Goal
Achieve 80%+ test coverage for classification loss functions.

## Labels
test-coverage, fitness, loss-functions, classification
